agent/battle/vision/parser.py

```
@@ -0,0 +1,125 @@
+    return VisualCard(slot, color, owner, _bool(value.get("is_np", False), "card.is_np"), _confidence(value.get("confidence", 0.0), "card.confidence"))
+
+def _servant(value: Any) -> VisualServant:
+    if not isinstance(value, dict):
+        raise VisionParseError("servant must be an object")
+    slot = _slot(value.get("slot"), 1, 3, "servant.slot")
+    skills = value.get("skill_available", [None, None, None])
+    if not isinstance(skills, list) or len(skills) != 3:
+        raise VisionParseError("servant.skill_available must contain 3 values")
+    np_percent = value.get("np_percent")
+    if np_percent is not None and (not isinstance(np_percent, int) or not 0 <= np_percent <= 300):
+        raise VisionParseError("servant.np_percent must be in [0, 300]")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)